### PR TITLE
Format specification API

### PR DESCRIPTION
**Pull request by hameerabbasi**

### CODE REVIEW

**Comment by mtsokol** [comment]
I see that the PR removes `reshape` and `broadcast_to`. Is it intended? I think these functions should stay as they are a part of Array API standard.

**Comment by hameerabbasi** [comment]
> I see that the PR removes `reshape` and `broadcast_to`. Is it intended? I think these functions should stay as they are a part of Array API standard.

I was planning on adding them back as part of a follow-up PR.

**Review by mtsokol — DISMISSED** [review]
I really like this change! LGTM!

Only two nitpick, and one comment.

**Comment by mtsokol** [comment]
> > I see that the PR removes `reshape` and `broadcast_to`. Is it intended? I think these functions should stay as they are a part of Array API standard.
> 
> I was planning on adding them back as part of a follow-up PR, with some heuristics on the format selection.

Ok, I agree that they can be added first after merging it.

File: sparse/mlir_backend/_array.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    def format(self) -> StorageFormat:
+        return self._storage.get_storage_format()
+
+    def _get_mlir_type(self):
+        return self.format._get_mlir_type(shape=self.shape)
+
+    def _to_module_arg(self):
+        return self._storage.to_module_arg()
+
+    def copy(self):
```
[the remark this answers is not in the document]
Comment (hameerabbasi) [reply sparse/mlir_backend/_array.py]: Done in https://github.com/pydata/sparse/pull/792/commits/f7162f8b12d0d0fb8e7e316249356f04ebe52c11.

File: sparse/mlir_backend/_array.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    @property
+    def shape(self) -> tuple[int, ...]:
+        return self._shape
+
+    @property
+    def ndim(self) -> int:
+        return len(self.shape)
+
+    @property
+    def dtype(self):
```
[the remark this answers is not in the document]
Comment (hameerabbasi) [reply sparse/mlir_backend/_array.py]: Done in https://github.com/pydata/sparse/pull/792/commits/f7162f8b12d0d0fb8e7e316249356f04ebe52c11.

**Review by mtsokol — APPROVED** [review]

**Pull request merged**